pkg/kubelet/cm/cgroup_manager.go

```diff
@@ -0,0 +1,114 @@
+// +build linux
+
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cm
+
+import (
+	"fmt"
+	"path"
+
+	libcontainercgroups "github.com/opencontainers/runc/libcontainer/cgroups"
+	cgroupfs "github.com/opencontainers/runc/libcontainer/cgroups/fs"
+	libcontainerconfigs "github.com/opencontainers/runc/libcontainer/configs"
+)
+
+// cgroupManagerImpl implements CgroupManager interface.
+// It Uses the Libcontainer raw fs cgroup manager for cgroup management.
+type cgroupManagerImpl struct {
+	// name of the container
+	name string
+	// CgroupConfig of the container
+	cgroupConfig *CgroupConfig
+	// Libcontainer raw fs cgroup manager.
+	fsCgroupManager libcontainercgroups.Manager
+}
+
+// Make sure that cgroupManagerImpl implements the CgroupManager interface
+var _ CgroupManager = &cgroupManagerImpl{}
+
+// NewCgroupManagerImpl is a factory method that returns
+// a configured CgroupManager
+func NewCgroupManagerImpl(cgroupConfig *CgroupConfig) (CgroupManager, error) {
```

Comment:
I don't think we need a cgroup manager per container. We only need one instance of cgroup manager to manage lifecycle of all cgroups..